metricbeat/module/system/process/process.go

```diff
@@ -0,0 +1,114 @@
+/*
+
+An example event looks as following:
+
+
+    {
+      "@timestamp": "2016-04-26T19:24:19.108Z",
+      "beat": {
+        "hostname": "ruflin",
+        "name": "ruflin"
+      },
+      "metricset": "process",
+      "module": "system",
+      "rtt": 20982,
+      "system-process": {
```

Comment:
Ok, those don't really work in Kibana, from what I know, but I guess there's not real reason for them not to work in a future UI. @simianhacker is the best to comment on that :-).

This can result in huge documents, not sure if we'll hit performance problems due to that, for example I read that it's not possible to return a singe nested object from Elasticsearch, you need to get the whole thing.

We also have to be aware that this will introduce a different kind of data that we'll have to take into account in other features. For example, generic filtering in its current form won't work on that, I think. But I worry more about the UI part, the nested and reverse-nested aggregations [here](https://www.elastic.co/guide/en/elasticsearch/guide/current/nested-aggregation.html) look pretty scary to me.

I'm not against this, btw, just making sure we understand what we're getting into :-).